src/intervals/intervals.jl

```diff
@@ -57,6 +54,38 @@ Interval{T}(x) where T = Interval(convert(T, x))
 
 Interval{T}(x::Interval) where T = convert(Interval{T}, x)
 
+"""
+    isvalid(a::Real, b::Real)
+
+Check if `(a, b)` constitute a valid interval
+"""
+function isvalid(a::Real, b::Real)
+    if isnan(a) || isnan(b)
+        return true
+    end
+
+    if a > b
+        if isinf(a) && isinf(b)
+            return true  # empty interval = [∞,-∞]
+        else
+            return false
+        end
+    end
+
+    return true
+end
+
+function interval(a::Real, b::Real)
+    if !isvalid(a, b)
+        throw(ArgumentError("Must have a ≤ b to construct interval(a, b)."))
+    end
+
+    return Interval(a, b)
+end
+
+interval(a::Real) = interval(a, a)
+
```

Comment:
If I understand `interval` *always* checks the validity of the construction, while `Interval` does not (currently as defualt) unless `IA_VALID=true`. Is this correct?